exonum/src/blockchain/mod.rs

```diff
@@ -287,15 +293,15 @@ impl BlockchainMut {
         }
 
         // Get tx & state hash.
-        let schema = Schema::new(&fork);
+        let schema = Schema::get_unchecked(&fork);
         let state_hash = {
             let mut sum_table = schema.state_hash_aggregator();
             // Clear old state hash.
             sum_table.clear();
             // Collect all state hashes.
             let state_hashes = self
                 .dispatcher
-                .state_hash(fork.as_ref())
+                .state_hash(fork.snapshot_with_flushed_changes())
```

Comment:
Is there any situation when user wants the snapshot to be without flushed changes?

Reply:
The idea is that `Fork` consists of 3 parts: an underlying `Snapshot`, flushed changes and unflushed changes. This method captures first 2 parts, but not the third one. I suppose `snapshot_without_unflushed_changes()` name could convey this more clearly.